tests/test_python_types.cpp

```diff
@@ -575,15 +575,18 @@ test_initializer python_types([](py::module &m) {
 
     m.def("return_capsule_with_name_and_destructor_3",
         []() {
-            py::print("creating capsule");
             auto capsule=py::object(py::capsule((void *) 1234, "pointer type description",
-                [](PyObject *ptr) {py::print("destructing capsule");}));
+                [](PyObject *ptr) {
+                 if(ptr){
+                    py::print("destructing capsule");
+                 }
+            }));
             auto name=PyCapsule_GetName(capsule.ptr());
-            py::print(name);
+            auto contents=PyCapsule_GetPointer(capsule.ptr(),name);
```

Comment:
It looks like you removed the `py::print("creating capsule");` but forgot to take that line out of the expected output in the test.